sdk/storage/azure-storage-file/azure/storage/file/_models.py

```diff
@@ -66,6 +75,13 @@ def __init__(self, enabled=False, days=None):
         if self.enabled and (self.days is None):
             raise ValueError("If policy is enabled, 'days' must be specified.")
 
+    @classmethod
+    def _from_generated(cls, generated):
```

Comment:
I think we need an `if generated:` check here - I think it can be `None`.
We may need "if None" checks in a number of the `_from_generated` methods as I'm not sure which properties are optional.

Reply:
good point, we can add that everywhere, and cannot hurt

Reply:
fixed